src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/ArgumentFromAttribute.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentValueResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\Argument\ArgumentMetadataInterface;
+
+/**
+ * Grabs a non-variadic value from the request and returns it.
+ *
+ * Opposite of {@see VariadicArgumentFromAttribute}.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class ArgumentFromAttribute implements ArgumentValueResolverInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadataInterface $argument)
+    {
+        return !$argument->isVariadic() && array_key_exists($argument->getArgumentName(), $request->attributes->all());
```

Comment:
Good point, that's a lot nicer. This is what happens if you "blindly" copy functionality :)

From:
`if (array_key_exists($param->name, $attributes)) {`